upstream.go

```diff
@@ -860,3 +1128,14 @@ func (uc *upstreamConn) readMessages(ch chan<- upstreamIncomingMessage) error {
 func (uc *upstreamConn) SendMessage(msg *irc.Message) {
 	uc.outgoing <- msg
 }
+
+func (uc *upstreamConn) SendMessageLabeled(dc *downstreamConn, msg *irc.Message) {
+	if uc.labelsSupported {
+		if msg.Tags == nil {
+			msg.Tags = make(map[string]irc.TagValue)
+		}
+		msg.Tags["label"] = irc.TagValue(fmt.Sprintf("sd-%d-%d", dc.id, uc.nextLabelId))
```

Comment:
Should this be obfuscated somehow? (I think not). Also, are uint64 counters OK for downstream ids and label ids? Either that or randomly generated UUIDs could work.

Reply:
Yeah, I think all of this is fine.